kikit/eeshema.py

```diff
@@ -0,0 +1,161 @@
+            header["revision"] = line.split(maxsplit=2)[1]
+        elif line.startswith("Comment1"):
+            header["comment1"] = line.split(maxsplit=2)[1]
+        elif line.startswith("Comment2"):
+            header["comment2"] = line.split(maxsplit=2)[1]
+        elif line.startswith("Comment3"):
+            header["comment3"] = line.split(maxsplit=2)[1]
+        elif line.startswith("Comment4"):
+            header["comment4"] = line.split(maxsplit=2)[1]
+        elif line.startswith("encoding"):
+            header["encoding"] = line.split(maxsplit=2)[1]
+        else:
+            raise EeschemaException(f"Unexpected line: '{line}'")
+
+def readComponent(file):
+    component = {
+        "reference": set()
+    }
+    while True:
+        line = readEeschemaLine(file)
+        if line == "$EndComp":
+            return component
+
+        if line.startswith("L"):
+            items = line.split()
+            component["reference"].add(items[2])
+            component["name"] = items[1]
+        elif line.startswith("U"):
+            component["u"] = line.split()[1:]
+        elif line.startswith("P"):
+            items = line.split()
+            component["position"] = (int(items[1]), int(items[2]))
+        elif line.startswith("F"):
+            items = shlex.split(line)
+            field = {
+                "number": int(items[1]),
+                "text": items[2],
+                "orientation": items[3],
+                "position": (int(items[4]), int(items[5])),
+                "size": items[6],
+                "flags": items[7],
+                "justify": items[8],
+                "style": items[9]
+            }
+            if field["number"] >= 4:
+                field["name"] = items[10]
+            component["fields"] = component.get("fields", []) + [field]
+        elif line.startswith("AR"):
+            # Hierarchical sheet reference. We assume all sheets are used within
+            # the project, therefore we do not check validity of path
+            items = shlex.split(line)
+            for item in items:
+                if item.startswith('Ref='):
+                    component["reference"].add(item[len('Ref='):])
+        else:
+            items = shlex.split(line)
+            try:
+                int(items[0])
+                if len(items) == 3 and items[0] == "1":
+                    # Duplicate position entry
+                    pass
+                if len(items) == 4:
+                    component["orientation"] = [int(x) for x in items[1:]]
+            except ValueError:
+                raise EeschemaException(f"Unexpected line: '{line}'")
+
+def readSheet(file):
+    # Note that the parser is incomplete an only extracts filename from the
+    # sheet as currently it is all we need
+    sheet = {}
+    while True:
+        line = readEeschemaLine(file)
+        if line == "$EndSheet":
+            return sheet
+        if line.startswith("F1"):
```

Comment:
```suggestion
        if line.startswith("F1 "):
```
Match the start of line exactly.  My design has "F10, F11, F12" and so on and this matches those as well.